pppd/sys-linux.c

```diff
@@ -127,9 +128,27 @@
 #include <linux/ppp_defs.h>
 #include <linux/if_ppp.h>
 
-#ifdef INET6
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#include <linux/if_link.h>
+/* Attempt at retaining compile-support with older than 4.7 kernels,
+ * older than 2.6.35 will probably break.  Either way, pre 4.7 kernels
+ * will not work with this code. */
+#ifndef RTM_NEWSTATS
+#define RTM_NEWSTATS 92
+#define RTM_GETSTATS 94
+#define IFLA_STATS_LINK_64 1
+#endif
+/* Since we only care about the initial few bytes, we use our own structure */
+struct rtnl_link_stats64_compat {
+    uint64_t rx_packets;
+    uint64_t tx_packets;
+    uint64_t rx_bytes;
+    uint64_t tx_bytes;
+};
+#define rtnl_link_stats64 rtnl_link_stats64_compat
```

Comment:
If you are not going to hide/guard this custom definition of `rtnl_link_stats64` then rather do not define it at all. And instead put it directly into `struct nlresp`. If it is possible it is better to not redefine or overwrite _standard_ structures to prevent possible future breakage. My original idea was to hide this `struct rtnl_link_stats64_compat` just for toolchains which are older (or do not define all constants) as these toolchains have less probability of breakage.

Reply:
Updated.